src/coreclr/jit/emitxarch.cpp

```diff
@@ -1309,18 +1332,43 @@ emitter::code_t emitter::AddEvexPrefix(const instrDesc* id, code_t code, emitAtt
 
     if (attr == EA_32BYTE)
     {
-        // Set L bit to 1 in case of instructions that operate on 256-bits.
+        // Set L bit to 01 in case of instructions that operate on 256-bits.
         code |= LBIT_IN_BYTE_EVEX_PREFIX;
     }
     else if (attr == EA_64BYTE)
     {
-        // Set L' bits to 11 in case of instructions that operate on 512-bits.
+        // Set L' bits to 10 in case of instructions that operate on 512-bits.
         code |= LPRIMEBIT_IN_BYTE_EVEX_PREFIX;
     }
 
     if (id->idIsEvexbContext())
     {
+
         code |= EVEX_B_BIT;
+
+        if (!id->idHasMem())
+        {
+            // embedded rounding case.
+            unsigned roundingMode = id->idGetEvexbContext();
+            if (roundingMode == 1)
+            {
+                // {rd-sae}
+                code &= ~(LPRIMEBIT_IN_BYTE_EVEX_PREFIX);
+                code |= LBIT_IN_BYTE_EVEX_PREFIX;
+            }
+            else if (roundingMode == 2)
+            {
+                // {ru-sae}
+                code |= LPRIMEBIT_IN_BYTE_EVEX_PREFIX;
+                code &= ~(LBIT_IN_BYTE_EVEX_PREFIX);
+            }
+            else
+            {
+                // {rz-sae}
+                code |= LPRIMEBIT_IN_BYTE_EVEX_PREFIX;
+                code |= LBIT_IN_BYTE_EVEX_PREFIX;
+            }
+        }
```

Comment:
Should we assert that the EVEX.B context is `1` in the `else` case? (Since anything else should have been invalid)

Reply:
Do you mean arrange the branches like:
```C#
if (id->idIsEvexbContext())
    {

        code |= EVEX_B_BIT;

        if (!id->idHasMem())
        {
            // embedded rounding case.
            unsigned roundingMode = id->idGetEvexbContext();
            if (roundingMode == 1)
            {
                // {rd-sae}
                code &= ~(LPRIMEBIT_IN_BYTE_EVEX_PREFIX);
                code |= LBIT_IN_BYTE_EVEX_PREFIX;
            }
            else if (roundingMode == 2)
            {
                // {ru-sae}
                code |= LPRIMEBIT_IN_BYTE_EVEX_PREFIX;
                code &= ~(LBIT_IN_BYTE_EVEX_PREFIX);
            }
            else if (roundingMode == 3)
            {
                // {rz-sae}
                code |= LPRIMEBIT_IN_BYTE_EVEX_PREFIX;
                code |= LBIT_IN_BYTE_EVEX_PREFIX;
            }
            else
            {
                // I am a bit unclear why we should assert Evex.b context is 1 here, 
                // other than the cases above, everything else should be invalid, 
                // so putting a `unreached()` might make more sense?
                // As we defined _idEvexbContext - 2 bits:
                // 00 - no embedded broadcast/rounding
                // 01 - embedded broadcast / round down
                // 10 - round up
                // 11 - round to zero
            }
        }
    }
```

Reply:
Basically what you did, but also covering the case where `id->idHasMem()` is true, so...
```cpp
        if (!id->idHasMem())
        {
            // embedded rounding case.
            unsigned roundingMode = id->idGetEvexbContext();
            if (roundingMode == 1)
            {
                // {rd-sae}
                code &= ~(LPRIMEBIT_IN_BYTE_EVEX_PREFIX);
                code |= LBIT_IN_BYTE_EVEX_PREFIX;
            }
            else if (roundingMode == 2)
            {
                // {ru-sae}
                code |= LPRIMEBIT_IN_BYTE_EVEX_PREFIX;
                code &= ~(LBIT_IN_BYTE_EVEX_PREFIX);
            }
            else if (roundingMode == 3)
            {
                // {rz-sae}
                code |= LPRIMEBIT_IN_BYTE_EVEX_PREFIX;
                code |= LBIT_IN_BYTE_EVEX_PREFIX;
            }
            else
            {
                unreached();
            }
        }
        else
        {
            assert(id->idGetEvexbContext() == 0);
        }
```